Source/Evolve/WarpXEvolve.cpp

```diff
@@ -571,7 +574,7 @@ WarpX::OneStep_multiJ (const amrex::Real cur_time)
         // Filter, exchange boundary, and interpolate across levels
         SyncRho();
         // Forward FFT of rho
-        PSATDForwardTransformRho(rho_fp, rho_cp, 0, 1);
+        PSATDForwardTransformRho(rho_fp, rho_cp, 0, rho_new);
```

Comment:
This is inside the `if` condition `if (rho_fp[0] && rho_in_time == RhoInTime::Linear)`, thus it is executed only when rho is linear in time, which means that the spectral index `rho_new` is well defined.